### PR TITLE
Header color change, fixes long list click

### LINKED ISSUE #364: Selecting and unselecting items
There are some minor issues regarding selecting and unselecting items:-

- On long pressing on items of **Flash Template** item is selected but item background color is not changed. Which will confuse user which item is selected.
videotogif_2017 01 02_20 08 31

- When items are selected and back button is pressed it finishes activity. On back button it should  unselect the items.

- As of now if we want to unselect an item we have to long press on it. I think it would we better to unselect on single click like in whatsapp.

### PR DESCRIPTION
**Pull request by akmittal006**
a UI update which includes following:

1. change of header color based on type of template selected. Provides more continuity and feel to app.
ezgif com-5da98cc0ca

2. Refactors some code

3. **fixes #364**

### CODE REVIEW

**Review by opticod — COMMENTED** [review]
@akmittal006, Thanks for these changes, but it looks like @vishwesh3 is also working for issue #364. Can you please remove the third fix from this PR, so that we can merge it. 

Note: Squash your commits into one , Thank you

**Comment by akmittal006** [comment]
@opticod it is done. 
Changes made - 
1. Header color change based on type of template
2. Refactoring of some code.

File: source-code/app/src/main/java/org/buildmlearn/toolkit/templates/ComprehensionAdapter.java [hunk 1]
```diff
@@ -42,7 +44,7 @@
 
     private static final int TYPE_HEADER = 0;
     private static final int TYPE_ITEM = 1;
-    private final Context context;
+    private static  Context context;
```
Comment (opticod) [inline source-code/app/src/main/java/org/buildmlearn/toolkit/templates/ComprehensionAdapter.java]: Using static field for Context variable will lead to memory leak. You can find many article on memory leak through static fields in web. 
Can you come up with different approach for this?

File: source-code/app/src/main/java/org/buildmlearn/toolkit/templates/MatchAdapter.java [hunk 2]
```diff
@@ -29,7 +33,7 @@
 
     private static final int TYPE_HEADER = 0;
     private static final int TYPE_ITEM = 1;
-    private final Context context;
+    private static  Context context;
```
Comment (opticod) [inline source-code/app/src/main/java/org/buildmlearn/toolkit/templates/MatchAdapter.java]: Same here!

**Review by opticod — CHANGES_REQUESTED** [review]
Nice effort! But couple of comments.

**Comment by akmittal006** [comment]
@opticod I have made the changes. Thanks for reviewing.

**Review by opticod — APPROVED** [review]
Thanks @akmittal006 Its looking as well as working great. :tada:

**Pull request merged**